pkg/db/src/__tests__/tests.ts

```diff
@@ -514,3 +514,57 @@ export const streamsShouldFallBackToDefaultValueForTheirType: TestFunction = asy
 		expect(n1UpdatedAt).toEqual(null);
 	});
 };
+
+export const BookInterface: TestFunction = async (db) => {
+	const book1 = {
+		isbn: '0195399706',
+
+		title: 'The Age of Wonder',
+		authors: 'Richard Holmes',
+		publisher: 'HarperCollins UK',
+		year: '2008',
+		price: 69.99
+	};
+	const book2 = {
+		isbn: '019976915X',
+
+		title: 'Twelve Bar Blues',
+		authors: 'Patrick Neate',
+		publisher: 'Penguin UK',
+		year: '2002',
+		price: 39.86
+	};
+
+	const bookInterface = db.books();
+
+	let bookEntries: (BookEntry | undefined)[] = [];
+
+	// eslint-disable-next-line @typescript-eslint/no-unused-vars
+	const [_, __, ___] = await Promise.all([
```

Comment:
Yeah, my initial intention was to write a test for the upsert as well as stream. I fixed it so that I first check for original book state then after it's updated.